<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:context="http://www.springframework.org/schema/context"
	xsi:schemaLocation="http://www.springframework.org/schema/beans 
        http://www.springframework.org/schema/beans/spring-beans-4.1.xsd
        http://www.springframework.org/schema/context 
        http://www.springframework.org/schema/context/spring-context-4.1.xsd">

	<import resource="classpath:/META-INF/spring/applicationContextTracage-dev.xml" />
	<context:property-placeholder location="classpath:META-INF/spring/*.properties" />

	<context:component-scan base-package="com.nordnet.topaze.contrat.domain" />
	<context:component-scan base-package="com.nordnet.topaze.contrat.controller" />
	<context:component-scan base-package="com.nordnet.topaze.contrat.repository" />
	<context:component-scan base-package="com.nordnet.topaze.contrat.service" />
	<context:component-scan base-package="com.nordnet.topaze.contrat.jms" />
	<context:component-scan base-package="com.nordnet.topaze.client.rest" />

	<!-- Load Hibernate related configuration -->
	<import resource="hibernate-context.xml" />

	<!-- Load activeMq related configuration -->
	<import resource="activemq-context.xml" />

	<!-- Load Quartez related configuration -->
	<import resource="quartz-context.xml" />

	<bean id="restURLProperties"
		class="org.springframework.beans.factory.config.PropertiesFactoryBean">
		<property name="location" value="classpath:rest-url.properties" />
	</bean>

	<!-- declaration du restTemplate avec un custom 'errorHandler' -->
	<bean class="org.springframework.web.client.RestTemplate">
		<property name="errorHandler">
			<bean class="com.nordnet.topaze.client.rest.RestClientResponseErrorHandler" />
		</property>
	</bean>

	<bean id="objectMapper" class="com.fasterxml.jackson.databind.ObjectMapper" />

	<bean id="topazeExceptionsProperties"
		class="org.springframework.beans.factory.config.PropertiesFactoryBean">
		<property name="location" value="classpath:topaze-exceptions.properties" />
	</bean>

	<bean id="dynamicProperties"
		class="org.springframework.context.support.ReloadableResourceBundleMessageSource">
		<!-- check property file(s) every 1 second -->
		<property name="cacheSeconds" value="1" />
		<property name="basenames">
			<list>
				<value>file:${dynamic.properties}</value>
			</list>
		</property>
	</bean>

	<!-- Bean d'acces au Contexte -->
	<bean id="applicationContextHolder"
		class="com.nordnet.topaze.contrat.util.spring.ApplicationContextHolder" />

	<bean id="log4jInitialization"
		class="org.springframework.beans.factory.config.MethodInvokingFactoryBean">
		<property name="targetClass" value="org.springframework.util.Log4jConfigurer" />
		<property name="targetMethod" value="initLogging" />
		<property name="arguments">
			<list>
				<value>classpath:/log4j-dev.xml</value>
			</list>
		</property>
	</bean>

	<bean id="sendAlert" class="com.nordnet.common.alert.ws.client.SendAlert">
		<property name="url">
			<value>${sendAlertServiceUrl}</value>
		</property>
	</bean>
</beans>